src/utils/react-query-factory.js

```diff
@@ -0,0 +1,56 @@
+// The keys that map to enterpriseQueryKeys that is used to define specific queries
+// and used to specifically invalidate queries
+
+export const queryKeys = {
+  // Global key to all queries
+  all: ['enterprise'],
+  enterprise: (enterpriseUuid) => [...queryKeys.all, enterpriseUuid],
+  enterpriseLearner: () => [...queryKeys.all, 'linked-enterprise-customer-users'],
+  // Global learner configuration
+  learnerConfiguration: (enterpriseUuid) => [...queryKeys.enterprise(enterpriseUuid), 'configuration'],
+  // Configurations
+  browseAndRequestConfiguration: (enterpriseUuid) => [...queryKeys.learnerConfiguration(enterpriseUuid), 'browse-and-request'],
+  enterpriseCurationConfiguration: (enterpriseUuid) => [...queryKeys.learnerConfiguration(enterpriseUuid), 'content-highlights'],
```

Comment:
* `browseAndRequestConfiguration` can be grouped with `allUserSubsidies`.
* `learnerConfiguration`... TBD if we need it?
* `enterpriseCurationConfiguration` -> can be moved to `contentHighlights` query key parent group
  * This leaves room for the future for a sub-query related to highlights to fetch `/highlight-sets` on the Search page route.

Reply:
All suggestions have been implemented. For the learner configuration I opted to make it feature specific and removed that attribute for now.